auditor/src/client/mod.rs

```diff
@@ -174,6 +175,139 @@ impl Default for AuditorClientBuilder {
     }
 }
 
+#[derive(serde::Deserialize, Debug, Default, Clone)]
+pub struct DateTimeUtcWrapper(pub DateTime<Utc>);
+
+impl Serialize for DateTimeUtcWrapper {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serializer.serialize_str(&self.0.to_rfc3339())
+    }
+}
+
+/// The 'QueryParameters' is used to build query parameters which allows to query records from
+/// the database using advanced_query function.
+#[derive(serde::Deserialize, serde::Serialize, Debug, Default, Clone)]
+pub struct QueryParameters {
+    pub start_time: Option<TimeOperator>,
+    pub stop_time: Option<TimeOperator>,
+    pub runtime: Option<TimeOperator>,
+}
+
+impl Default for QueryBuilder {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+#[derive(serde::Deserialize, Debug, Clone)]
+pub enum TimeValue {
+    Datetime(DateTimeUtcWrapper),
+    Runtime(u64),
+}
+
+impl Serialize for TimeValue {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        match self {
+            TimeValue::Datetime(datetime) => datetime.serialize(serializer),
+            TimeValue::Runtime(runtime) => runtime.serialize(serializer),
+        }
+    }
+}
+
+/// The 'TimeOperator' struct is used to specify the operators on the query parameters like (gt ->
+/// greater than, lt -> lesser than, gte -> greater than equals, lte -> lesser than equals)
+#[derive(serde::Deserialize, serde::Serialize, Debug, Default, Clone)]
+pub struct TimeOperator {
+    pub gt: Option<TimeValue>,
+    pub lt: Option<TimeValue>,
+    pub gte: Option<TimeValue>,
+    pub lte: Option<TimeValue>,
+}
+
+impl TimeOperator {
+    pub fn gt(mut self, value: TimeValue) -> Self {
+        self.gt = Some(value);
+        self
+    }
+
+    pub fn lt(mut self, value: TimeValue) -> Self {
+        self.lt = Some(value);
+        self
+    }
+
+    pub fn gte(mut self, value: TimeValue) -> Self {
+        self.gte = Some(value);
+        self
+    }
+
+    pub fn lte(mut self, value: TimeValue) -> Self {
+        self.lte = Some(value);
+        self
+    }
+}
+
+impl From<chrono::DateTime<Utc>> for TimeValue {
+    fn from(item: chrono::DateTime<Utc>) -> Self {
+        TimeValue::Datetime(DateTimeUtcWrapper(item))
+    }
+}
+
+impl From<u64> for TimeValue {
+    fn from(item: u64) -> Self {
+        TimeValue::Runtime(item)
+    }
+}
+
+/// The Builder to automatically build the 'QueryParameter' struct using builder pattern
+#[derive(Debug, Clone)]
+pub struct QueryBuilder {
+    pub query_params: QueryParameters,
+}
+
+impl QueryBuilder {
```

Comment:
Documentation: Please add a short description for each public function (again similar to the AuditorClientBuilder)